packages/roosterjs-editor-plugins/lib/plugins/ImageSelection/ImageSelection.ts

```diff
@@ -37,35 +45,47 @@ export default class ImageSelection implements EditorPlugin {
     dispose() {
         this.editor.select(null);
         this.editor = null;
-        this.imageId = null;
     }
 
     onPluginEvent(event: PluginEvent) {
         if (this.editor) {
             switch (event.eventType) {
                 case PluginEventType.EnteredShadowEdit:
+                case PluginEventType.LeavingShadowEdit:
                     const selection = this.editor.getSelectionRangeEx();
                     if (selection.type == SelectionRangeTypes.ImageSelection) {
                         this.editor.select(selection.image);
                     }
                     break;
-                case PluginEventType.LeavingShadowEdit:
-                    if (this.imageId) {
-                        const images = this.editor.queryElements(
-                            '#' + this.imageId
-                        ) as HTMLImageElement[];
-                        if (images.length == 1) {
-                            const image = images[0];
-                            this.editor.select(image);
-                        }
-                        this.imageId = null;
-                    }
-                    break;
+
                 case PluginEventType.MouseDown:
                     const target = event.rawEvent.target;
                     if (safeInstanceOf(target, 'HTMLImageElement')) {
                         this.editor.select(target);
-                        this.imageId = target.id;
+                    }
+                    break;
+                case PluginEventType.KeyDown:
+                    const key = event.rawEvent.key;
+                    const keyDownSelection = this.editor.getSelectionRangeEx();
+
+                    if (keyDownSelection.type === SelectionRangeTypes.ImageSelection) {
+                        if (key === Escape) {
+                            this.editor.select(keyDownSelection.image, PositionType.Before);
+                            this.editor.getSelectionRange().collapse();
+                        } else if (
+                            key === SHIFT ||
+                            key === CTRL ||
+                            key === Alt ||
+                            event.rawEvent.metaKey
+                        ) {
+                            this.editor.select(keyDownSelection.ranges[0]);
+                        } else if (key === Backspace || key === Enter) {
+                            event.rawEvent.preventDefault();
```

Comment:
Hmm, for Backspace and Enter (as well as Delete), we should remove the selected image (and add a new line for Enter key). So better do the same handling with Ctrl/Shift/Alt to select the image then let browser handle the deleting for us.

Reply:
I was copying the before word online that nothing happen for theses keys. Backspace is a type, is space only. I'll change.